modules/processors/multidomain_dataset_processor.py

```diff
@@ -444,3 +536,411 @@ def chunk_text(text, title, max_size=1000, overlap=200):
             all_chunks.extend(chunks)
         dataset = datasets.Dataset.from_pandas(pd.DataFrame(all_chunks))
         return dataset
+
+
+class MultiQA(Processor):
+    """
+    This dataset contains a combination of the following QA datasets:
+
+        < QUERY >                     |     < DOCS >
+        nq open                       |     odqa-wiki-corpora-100w-karpukhin ~ kilt-100w
+        msmarco 2.1 (first 100k)      |     ms-marco on huggingface = irds/msmarco-passage ~ ms-marco_full on bergen
+        adverserial qa                |     squad ~ wikipedia ~ kilt-100w
+        hotpotqa                      |     kilt-100w
+        wikiqa                        |     kilt-100w
+        sciq                          |     kilt-100w
+        asqa                          |     kilt-100w
+        triviaqa                      |     kilt-100w
+        freebase_qa                   |     freebase extract from https://github.com/kelvin-jiang/FreebaseQA (not integrated in bergen)
+        squad_v1.1                    |     wikipedia ~ kilt-100w
+    """
+    def __init__(self, *args, **kwargs):
+        dataset_name = 'MultiQA'
+        super().__init__(*args, **kwargs, dataset_name=dataset_name)
+
+    def process(self):
+        ds = datasets.load_dataset("dmrau/multi_qa", num_proc=self.num_proc)[self.split]
+        return ds
+    
+
+class TechQA(Processor):
+    """
+    Paper: https://aclanthology.org/2020.acl-main.117.pdf
+    Official source: https://github.com/IBM/techqa/tree/master/docker/techqa
+    Source we use: https://huggingface.co/datasets/rojagtap/tech-qa
+    Note: we combine train/validation/test splits to have a bigger dev dataset
+    """
+    def __init__(self, *args, **kwargs):
+        dataset_name = 'TechQA'
+        super().__init__(*args, **kwargs, dataset_name=dataset_name)
+
+    def process(self):
+        ds = datasets.load_dataset("rojagtap/tech-qa")
+        dataset = datasets.concatenate_datasets([ds["train"], ds["validation"], ds["test"]])
+        def map_fn(example):
+            example['label'] = [example['answer']]
+            return example
+        dataset = dataset.map(map_fn, num_proc=self.num_proc)
+        dataset = dataset.rename_column("question", "content")
+        dataset = dataset.remove_columns(["document", "answer"])
+        return dataset
+
+class TechQA_docs(Processor):
+    def __init__(self, *args, **kwargs):
+        dataset_name = 'TechQA_docs'
+        super().__init__(*args, **kwargs, dataset_name=dataset_name)
+
+    def process(self):
+        ds = datasets.load_dataset("rojagtap/tech-qa")
+        dataset = datasets.concatenate_datasets([ds["train"], ds["validation"], ds["test"]])
+
+        def chunk_text(text, title, id, max_size=1000, overlap=200):
```

Comment:
Can't you re-use the chunk_text method that you added in utils there ?
Also: type hints :)

Reply:
done